### PR TITLE
Support SSO sign in for CodeCatalyst in IDEA

### PR DESCRIPTION
**Pull request by juanbenj**
This commit adds support in IDEA for signing in to CodeCatalyst via the AWS IAM Identity Center

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/sono/CodeCatalystCredentialManager.kt [hunk 1]
```diff
@@ -63,40 +64,38 @@ class SonoCredentialManager {
...
-                loginSso(project, SONO_URL, SONO_REGION, CODECATALYST_SCOPES)
-            }
+    internal fun promptAuth(): BearerTokenProvider {
+        connection()?.let {
+            return reauthProviderIfNeeded(project, provider(it), isBuilderId = true)
+        }
 
-            else -> reauthProviderIfNeeded(project, provider, isBuilderId = true)
+        return runUnderProgressIfNeeded(project, message("credentials.pending.title"), true) {
+            if (requestCredentialsForCodeCatalyst(project as Project)) {
```
[the remark this answers is not in the document]
Comment (juanbenj) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/sono/CodeCatalystCredentialManager.kt:79]: The type of `project` is `Project?`, but the function expects a `Project`. This is more of a temporary stopgap measure to obtain a working feature. Long term, I recommend we work towards making `project` optional where needed/applicable.

An alternative is to pass a dummy project, but that seems much worse in my opinion.

[the remark this answers is not in the document]
Comment (manodnyab) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/sono/CodeCatalystCredentialManager.kt:79]: What if wrap the call in a null project check?

[the remark this answers is not in the document]
Comment (rli) [reply jetbrains-core/src/software/aws/toolkits/jetbrains/core/credentials/sono/CodeCatalystCredentialManager.kt:79]: passing the null all the way down is the right way to do this

File: jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/GettingStartedAuthUtils.kt [hunk 2]
```diff
@@ -156,6 +157,91 @@ fun requestCredentialsForCodeWhisperer(
...
+        scopes = CODECATALYST_SCOPES,
+        promptForIdcPermissionSet = false,
+        sourceOfEntry = SourceOfEntry.CODECATALYST,
+        featureId = FeatureId.Codecatalyst,
+        isFirstInstance = isFirstInstance,
+        connectionInitiatedFromExplorer = connectionInitiatedFromExplorer
+    )
+    val isAuthenticationSuccessful = authenticationDialog.showAndGet()
+    if (isAuthenticationSuccessful) {
+        AuthTelemetry.addConnection(
```
Comment (manodnyab) [inline jetbrains-core/src/software/aws/toolkits/jetbrains/core/gettingstarted/GettingStartedAuthUtils.kt:212]: For Telemetry, we will also need to update the function: `getEnabledConnectionsForTelemetry` to reflect whether the enabled connections are Idc or Builder Id

**Comment by rli** [comment]
https://github.com/aws/aws-toolkit-jetbrains/pull/4077

**Pull request closed without merging**